File: level1JS/delSmallestNum.js
function solution(arr) {
    if(arr.length === 1) {
        return [-1];
    }

    let smallestNum = 0;
    let targetIndex = 0;

    arr.forEach((num, index) => {
        if(index === 0) return smallestNum = num;

        if(smallestNum > num) {
            smallestNum = num;
            targetIndex = index;
        }
    })
    arr.splice(targetIndex, 1);
    return arr;
}

console.log(solution([4,3,2,1]))

File: level1JAVA/src/DelSmallestNum.java
import java.util.Arrays;

class DelSmallestNum {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length - 1];
        int[] temp = Arrays.copyOf(arr, arr.length);
        if(arr.length == 1) return new int[]{-1};
        Arrays.sort(temp);
        System.out.println("temp = " + Arrays.toString(temp));
        int smallestNum = temp[0];
        int index = 0;
        for(int num : arr) {
            if(num == smallestNum) continue;
            answer[index] = num;
            index++;
        }
        return answer;
    }

    public static void main(String[] args) {
        DelSmallestNum delSmallestNum = new DelSmallestNum();
        int[] result = delSmallestNum.solution(new int[]{2,5,1,2,3,41});
        System.out.println("result = " + Arrays.toString(result));
    }
}
